spdx-utils/build.gradle.kts

```diff
@@ -65,8 +65,10 @@ rootProject.idea {
 dependencies {
     antlr("org.antlr:antlr4:$antlrVersion")
 
-    implementation("com.fasterxml.jackson.core:jackson-databind:$jacksonVersion")
-    implementation("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:$jacksonVersion")
+    api("com.fasterxml.jackson.core:jackson-databind:$jacksonVersion")
+    api("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:$jacksonVersion")
```

Comment:
> Are the createYamlMapper and createJsonMapper functions the only reason to make these api dependencies?

not quite sure I have to admit.

> I think the functions could be made internal

That's not possible as it would lead to a compile error pointing to the two `reified` functions in that file.

Reply:
I thought a bit about this because I would like it if we could keep them implementation dependencies, the only option I found that also keeps the function generic is this:

```
    internal fun createJsonMapper(): ObjectMapper = JsonMapper().apply(mapperConfig)
    fun <T : Any> T.fromJson(yaml: String): T = createJsonMapper().readValue(yaml, this.javaClass)
```

Reply:
Hm, but how would the caller code look like then? ...because it becomes an extension function then and now it is in the Serializer object.